### PR TITLE
ADD custom methods

### PR DESCRIPTION
**Pull request by emmanuellmhango**
I have added the following

- Create model classes for all entities as shown in the.
- Set up associations between models.
 (author_id column in posts table should be the foreign_key for the users table.)
- Create custom methods

sec
sec2
sec3
sec4
insert users
insert users2

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :comments
+  has_many :posts
+  has_many :likes
```
Comment (franclobo) [inline app/models/user.rb]: - [ ] Kindly add 👉🏽 `foreign_key: 'author_id'` for pos,t comments, and likes to match the given [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
```
Comment (franclobo) [inline app/models/post.rb]: - [ ] Please 🙏🏽 remove the foreign key here, it should be added to the user model. The user has many posts then the user needs a foreign key to join with the post table.

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Comment < ApplicationRecord
+  belongs_to :post
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
```
Comment (franclobo) [inline app/models/comment.rb]: - [ ] Please 🙏🏽 remove the foreign key here, it should be added to the user model. As I commented in the post.

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class Like < ApplicationRecord
+  belongs_to :post
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
```
Comment (franclobo) [inline app/models/like.rb]: - [ ] Please 🙏🏽 remove the foreign key here, it should be added to the user model. As I commented in the post.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @emmanuellmhango, 

# ♻️ Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct methods ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ismailco — APPROVED** [review]
### Hi @emmanuellmhango, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**